src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonAPI.kt

```diff
@@ -105,17 +105,17 @@ object DungeonAPI {
     }
 
     private fun checkBossName(bossName: String): Boolean {
-        val correctBoss = when (dungeonFloor!!) {
-            "E" -> "The Watcher"
-            "F1", "M1" -> "Bonzo"
-            "F2", "M2" -> "Scarf"
-            "F3", "M3" -> "The Professor"
-            "F4", "M4" -> "Thorn"
-            "F5", "M5" -> "Livid"
-            "F6", "M6" -> "Sadan"
-            "F7", "M7" -> "Maxor"
-            else -> null
-        } ?: return false
+        val floor = dungeonFloor ?: return false
+        val correctBoss = when (floor) {
+            DungeonFloor.E -> "The Watcher"
+            DungeonFloor.F1, DungeonFloor.M1 -> "Bonzo"
+            DungeonFloor.F2, DungeonFloor.M2 -> "Scarf"
+            DungeonFloor.F3, DungeonFloor.M3 -> "The Professor"
+            DungeonFloor.F4, DungeonFloor.M4 -> "Thorn"
+            DungeonFloor.F5, DungeonFloor.M5 -> "Livid"
+            DungeonFloor.F6, DungeonFloor.M6 -> "Sadan"
+            DungeonFloor.F7, DungeonFloor.M7 -> "Maxor"
```

Comment:
```suggestion
            DungeonFloor.F7, DungeonFloor.M7 -> "Maxor"
            else -> floor.bossName
```

Reply:
i'm confused about this one

Reply:
Just simplifying the code. It's different for F7/M7 because it checks for the first boss you fight (Maxor) instead of the final boss (Necron).